src/Sentry/SentryClient.cs

```diff
@@ -102,27 +101,27 @@ public SentryId CaptureEvent(SentryEvent @event, Scope scope = null)
         /// <returns>A task to await for the flush operation.</returns>
         public Task FlushAsync(TimeSpan timeout) => Worker.FlushAsync(timeout);
 
-        private SentryId DoSendEvent(SentryEvent @event, Scope scope)
+        // TODO: this method needs to be refactored, it's really hard to analyze nullability
+        private SentryId DoSendEvent(SentryEvent @event, Scope? scope)
         {
-            if (_options.SampleRate is float sample)
+            if (_options.SampleRate != null)
```

Comment:
Any specific reason to change this? Not sure why change from the pattern matching to null check and a call on the property again.

One of the reasons I prefer the pattern matching is that the value is available on the reference declared, as opposed to a second access to property, that could possibly have change in between the two reads if mutable.

I realize the amount of bikesheeding that C# has created by having so many ways to do the same thing